core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/DatastoreServerBootstrap.java

```diff
@@ -147,9 +151,26 @@ public void run(SnowOwlConfiguration configuration, Environment env, IProgressMo
 			initializeJobSupport(env, configuration);
 			initializeRepositories(configuration, env);
 			initializeRequestSupport(env, configuration.getModuleConfig(RepositoryConfiguration.class).getNumberOfWorkers());
+			verifyRepositories(env);
 		}
 	}
 	
+	private void verifyRepositories(Environment env) {
+		final DefaultRepositoryManager repositories = (DefaultRepositoryManager) env.service(RepositoryManager.class);
+		Optional<InternalRepository> inconsistentRepository = repositories.repositories()
+			.stream()
+			.filter(repository -> repository instanceof InternalRepository).map(InternalRepository.class::cast)
+			.filter(repository -> !repository.isConsistent(BranchPathUtils.createMainPath()))
+			/* XXX: we are checking MAIN only */
+			.findAny();
+		
+		if (inconsistentRepository.isPresent()) {
+			String message = String.format("Inconsistent repository: %s.", inconsistentRepository.get().getCdoRepository().getRepositoryName());
+			throw new SnowOwlApplication.InitializationException(message);
```

Comment:
Unfortunately, the current implementation prevents users/admins to perform `reindex` or other diagnostic/maintenance operations, since they depend on the services initialized by the boot process. Let's chat about this issue when you have time, @ekovacs, @apeteri!